internal/handlers/pg/msg_count.go

```diff
@@ -50,26 +51,40 @@ func (h *Handler) MsgCount(ctx context.Context, msg *wire.OpMsg) (*wire.OpMsg, e
 		Collection: params.Collection,
 	}
 
-	qp.Filter = params.Filter
+	if !h.DisableFilterPushdown {
+		qp.Filter = params.Filter
+	}
 
 	var resDocs []*types.Document
 	err = dbPool.InTransaction(ctx, func(tx pgx.Tx) error {
-		resDocs, err = fetchAndFilterDocs(ctx, &fetchParams{tx, &qp, h.DisableFilterPushdown})
-		return err
+		var iter types.DocumentsIterator
+
+		iter, _, err = pgdb.QueryDocuments(ctx, tx, &qp)
+		if err != nil {
+			return err
```

Comment:
Should this one be a lazy error?

Reply:
`QueryDocuments` already returns lazyerror

Reply:
Nested lazy errors help to trace the chain of exact error calls, so it might make sense to return them everywhere where they are needed. Then in case they happen, it will be easier to debug the chain of errors.

Reply:
Otherwise, that would be a broken chain :)